roles/tests/test_api.py

```diff
@@ -280,3 +283,105 @@ def test_clear_permissions(self):
+            [
+                (self.user, BaseGroupTypes.ADMINISTRATORS)
+            ]
+        )
+        # Remove the user from the administrators.
+        api.remove_user_from_repo_group(
+            self.user,
+            self.repo,
+            GroupTypes.REPO_ADMINISTRATOR
+        )
+        # Add user to the curators.
+        api.assign_user_to_repo_group(
+            self.user,
+            self.repo,
+            GroupTypes.REPO_CURATOR
+        )
+        self.assertListEqual(
+            api.list_users_in_repo(self.repo),
+            [
+                (self.user, BaseGroupTypes.CURATORS)
+            ]
+        )
+        # Add user back to the administrators.
+        api.assign_user_to_repo_group(
+            self.user,
+            self.repo,
+            GroupTypes.REPO_ADMINISTRATOR
+        )
+        self.assertListEqual(
+            sorted(
+                api.list_users_in_repo(self.repo),
+                key=operator.itemgetter(1)
+            ),
+            sorted(
+                [
+                    (self.user, BaseGroupTypes.ADMINISTRATORS),
+                    (self.user, BaseGroupTypes.CURATORS)
+                ],
+                key=operator.itemgetter(1)
+            )
+        )
+        # Add another user to the authors.
+        api.assign_user_to_repo_group(
+            self.user_norepo,
+            self.repo,
+            GroupTypes.REPO_AUTHOR
+        )
+        self.assertListEqual(
+            sorted(
+                api.list_users_in_repo(self.repo),
+                key=operator.itemgetter(1)
+            ),
+            sorted(
+                [
+                    (self.user, BaseGroupTypes.ADMINISTRATORS),
+                    (self.user, BaseGroupTypes.CURATORS),
+                    (self.user_norepo, BaseGroupTypes.AUTHORS)
+                ],
+                key=operator.itemgetter(1)
+            )
+        )
+
+    def test_list_users_in_repo_base_group_type_specified(self):
+        """
+        Test for list_users_in_repo
+        """
+        self.assertListEqual(
+            api.list_users_in_repo(self.repo),
+            [
+                (self.user, BaseGroupTypes.ADMINISTRATORS)
+            ]
+        )
+        self.assertListEqual(
+            api.list_users_in_repo(self.repo, BaseGroupTypes.ADMINISTRATORS),
+            [
+                (self.user, BaseGroupTypes.ADMINISTRATORS)
+            ]
+        )
+        self.assertListEqual(
+            api.list_users_in_repo(self.repo, BaseGroupTypes.CURATORS),
+            []
+        )
+        self.assertListEqual(
+            api.list_users_in_repo(self.repo, BaseGroupTypes.AUTHORS),
+            []
+        )
+
+    def test_list_users_in_repo_invalid_group_type_specified(self):
+        """
+        Test for list_users_in_repo
+        """
+        self.assertRaises(
+            api.InvalidGroupType,
+            lambda: api.list_users_in_repo(self.repo, 'foo_type')
+        )
```

Comment:
For some reason I thought this was a different file. Yes I meant the same file we're commenting on :)